### PR TITLE
chore(build): Switch from yarn3 to npm

### PR DESCRIPTION
**Pull request by vovsemenv**
Sucessor of https://github.com/allure-framework/allure-js/pull/471 with fixed versions in publish workflow

How it works:
Now we have version updater script https://github.com/allure-framework/allure-js/pull/562/files#diff-20f13632b1834f7768ac8989d7ab6bb117ae1cec025befd04e4891b7aca335ba that will set versions of workspace deps

### CODE REVIEW

**Comment by vovsemenv** [comment]
I tested releasy cycle here https://github.com/vovsemenv/allure-js
version updates works fine

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,30 +19,46 @@ jobs:
       matrix:
         node-version: [14.x, 16.x]
     steps:
-      - uses: actions/checkout@v3.1.0
```
Comment (baev) [inline .github/workflows/build.yml:22]: why downgrade

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -19,30 +19,46 @@ jobs:
...
+      - uses: actions/checkout@v2.4.2
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
```
Comment (baev) [inline .github/workflows/build.yml:25]: why downgrade

File: packages/allure-jest/package.json [hunk 3]
```diff
@@ -19,17 +19,15 @@
     "compile": "tsc",
     "generate-report": "allure generate ./out/allure-results -o ./out/allure-report --clean",
     "jasmineTest": "jest --config ./test/default-runner/jest.config.js",
-    "test": "rimraf ./out; yarn run jasmineTest"
+    "test": "rimraf ./out; npm run jasmineTest"
```
Comment (baev) [inline packages/allure-jest/package.json:22]: use preTest instead

File: packages/allure-jasmine/package.json [hunk 4]
```diff
@@ -17,18 +17,16 @@
     "lint:fix": "eslint ./src ./index.ts --ext .ts --fix",
     "compile": "tsc",
     "generate-report": "allure generate ./out/allure-results -o ./out/allure-report --clean",
-    "runTest": "tsc -p tsconfig-tests.json && yarn run jasmine ./dist-tests/test/*.js",
+    "runTest": "tsc -p tsconfig-tests.json && npm run jasmine ./dist-tests/test/*.js",
     "runTestDebug": "node $NODE_DEBUG_OPTION ./node_modules/.bin/jasmine ./dist/test/*.js",
-    "test": "rimraf ./out ; yarn run runTest ; yarn run generate-report"
+    "test": "rimraf ./out ; npm run runTest ; npm run generate-report"
```
Comment (baev) [inline packages/allure-jasmine/package.json:22]: use preTest instead

File: update-local-deps.mjs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { readFile, writeFile } from "node:fs/promises";
```
Comment (baev) [inline update-local-deps.mjs:1]: Maybe use relative package links and add a pre-package script that replaces relative versions with the correct ones?

**Review by baev — CHANGES_REQUESTED** [review]
- [ ] Ensure package contents (attach the contents of tarballs to the PR)
- [ ] Ensure package versions are correct when run `npm pack`
- [ ] Add instructions on how to add a new package
- [ ] Add instructions on how to work with this workspace setup (how to build, how to test changes in commons module, etc)

**Pull request closed without merging**